VideoLocker/src/main/java/org/edx/mobile/discussion/DiscussionTextUtils.java

```diff
@@ -13,21 +13,43 @@
     private DiscussionTextUtils() {
     }
 
-    public static String getAuthorAttributionText(@NonNull IAuthorData authorData, @NonNull Resources resources) {
-        String text = authorData.getAuthor() + " " + DateUtils.getRelativeTimeSpanString(
+    public enum AuthorAttributionType {
+        DISCUSSION,
+        QUESTION,
+        COMMENT
+    }
+
+    public static CharSequence getAuthorAttributionText(@NonNull IAuthorData authorData, @NonNull Resources resources, @NonNull AuthorAttributionType attributionType) {
+        @StringRes int stringRes;
+        switch (attributionType) {
+            case DISCUSSION:
+                stringRes = R.string.discussion_thread_attribution;
```

Comment:
Could we have the `stringRes` variable defined as a property inside the enum itself? That might be more elegant, especially since that is the purpose of defining the enum in the first place.